### PR TITLE
add facts api to go sdk

### PR DESCRIPTION
**Pull request by RazcoDev**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/permitio/permit-golang
 
-go 1.17
+go 1.18
```
Comment (omer9564) [inline go.mod:3]: verify it doesn't break clients with go 1.17

File: pkg/api/relationshipTuples.go [hunk 2]
```diff
@@ -7,21 +7,28 @@ import (
...
+			permitBaseApi{
+				client: client,
+				config: config,
+				logger: config.Logger,
+			},
 		},
 	}
 }
+func (u *RelationshipTuples) WaitForSync(timeout *time.Duration) *RelationshipTuples {
+	u.PermitBaseFactsApi.WaitForSync(timeout)
```
Comment (omer9564) [inline pkg/api/relationshipTuples.go:29]: PermitBaseFactsApi.WaitForSync returns a new client that you initialized inside the function, but you doesn't return it from this function ( relevant for all the functions ).
This means that whatever WaitForSync is doing is currently unused

Comment (omer9564) [reply pkg/api/relationshipTuples.go:29]: If I'm correct you should do return NewRelationshipTuplesApi(u.PermitBaseFactsApi.WaitForSync(timeout)) or something like this

**Review by omer9564 — CHANGES_REQUESTED** [review]

File: pkg/api/relationshipTuples.go (hunk 2 above)
Comment (RazcoDev) [reply pkg/api/relationshipTuples.go:29]: true. fixed

File: go.mod (hunk 1 above)
Comment (RazcoDev) [reply go.mod:3]: that was a try of using generics, i'm reverting it to 1.17

**Review by omer9564 — APPROVED** [review]
LGTM

**Pull request merged**